docs/docs/reference/other-new-features/tupled-function.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: doc-page
+title: "Tupled Function"
+---
+
+Tupled Function
+----------------------
+
+With functions bounded to arities up to 22 it was possible to generalize some operation on all function types using overloading. 
+Now that we have functions and tuples generalized to [arities above 22](https://dotty.epfl.ch/docs/reference/dropped-features/limit22.html) overloading is not an option anymore. 
+The type class `TupleFunction` provides a way to abstract directly over functions of any arity converting it to an equivalent function that receives all arguments in a single tuple.
+
+```scala
+/** Type class relating a `FunctionN[..., R]` with an equvalent tupled function `Function1[TupleN[...], R]`
+ *
+ *  @tparam F a function type
+ *  @tparam G a tupled function type (function of arity 1 receiving a tuple as argument)
+ */
+@implicitNotFound("${F} cannot be tupled as ${G}")
+sealed trait TupledFunction[F, G] {
+  def tuple(f: F): G
+}
+```
+
+The compiler will synthesize an instance of `TupledFunction[F, G]` if:
+
+* `F` is a function type of arity `N`
+* `G` is a function with a single tuple argument of size `N` and it's types are equal to the arguments of `F`
+* The return type of `F` is equal to the return type of `G`
+* `F` and `G` are the same kind of functions (`given` arguments or not)
```

Comment:
This line is not clear to me, can you elaborate, please?

Reply:
If both are normal functions `FunctionN` or both are `ImplicitFunctionN`. How would you stat it?

Reply:
```suggestion
* both `F` and `G` have type `FunctionN` or they both have type `ImplicitFunctionN`
```
?

Reply:
I updated the documentation. Is it clearer now?

Reply:
Now I used
*  `F` and `G` are the same kind of function (both are `(...) => R` or both are `given (...) => R`)